aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -471,11 +486,26 @@ bool use_mkldnn_bf32_matmul(
       checksize(mat1, mat2));
 }
 
+bool use_mkldnn_tf32_matmul(
+    const Tensor& mat1,
+    const Tensor& mat2,
+    const Tensor& result) {
+
+    return (
+      use_mkldnn_tf32_matmul() &&
+      mat1.scalar_type() == kFloat &&
+      mat2.scalar_type() == kFloat &&
+      (!result.defined() || result.scalar_type() == kFloat) &&
+      mat1.numel() != 0 &&
+      mat2.numel() != 0 &&
+      checksize(mat1, mat2));
+}
+
```

Comment:
this file has multiple functions that have similar usage:
`use_mkldnn_bf16_matmul`, `use_mkldnn_fp16_matmul`, `use_mkldnn_bf32_matmul` and `use_mkldnn_tf32_matmul`

can we templatize it to simplify the code?

```
template <typename T>
bool use_mkldnn_matmul();

#if defined(__aarch64__)
bool use_mkldnn_matmul<at::BFloat16>();
#endif
```